src/mesh/unstructured_mesh.C

```diff
@@ -55,6 +55,62 @@ UnstructuredMesh::UnstructuredMesh (const Parallel::Communicator & comm_in,
   libmesh_assert (libMesh::initialized());
 }
 
+MeshBase & UnstructuredMesh::operator= (MeshBase && other_mesh)
+{
+  // First make a call to MeshBase::clear
+  this->MeshBase::clear();
+
+  this->ParallelObject::operator=(other_mesh);
+
+  // First set this mesh equal to other_mesh as a MeshBase.
+  this->MeshBase::operator=(std::move(other_mesh));
+
+  this->copy_nodes_and_elements(dynamic_cast<UnstructuredMesh &>(other_mesh), true);
```

Comment:
This is adding an RTTI dependency; that's a no.

Also: we're trying to do an O(1) move here, not an O(N) copy.  Using an API with "copy" in the name is a good sign that we're failing.  ;-)  To actually do a move, we're going to have to do it in the subclasses with the containers whose contents we're wanting to move.

Reply:
Oops, guess I got so caught up in boundary functors and virtual functions that I forgot all about the nodes and elements .... Will use the move assignment on the _elements and _nodes vectors.